internal/context/init.go

```diff
@@ -38,31 +38,55 @@ func Init(projectName string, outDir string) *projectconfig.ZeroProjectConfig {
 	projectConfig.Context["ShouldPushRepoUpstream"] = promptPushRepoUpstream()
 	projectConfig.Context["GithubRootOrg"] = promptGithubRootOrg()
 	projectConfig.Context["githubPersonalToken"] = promptGithubPersonalToken(projectName)
-	projectConfig.Modules = chooseStack(getRegistry())
 
 	// chooseCloudProvider(&projectConfig)
 	// fmt.Println(&projectConfig)
 	// s := project.GetSecrets(rootDir)
 	// fillProviderDetails(&projectConfig, s)
 	// fmt.Println(&projectConfig)
+	moduleSources := chooseStack(getRegistry())
+	moduleConfig := loadAllModules(moduleSources)
+	for _ = range moduleConfig {
+		// TODO: initialize module structs inside project
+	}
+
+	projectParameters := promptAllModules(moduleConfig)
+	for _ = range projectParameters {
+		// TODO: Add parameters to module structs inside project
+	}
```

Comment:
Being able to call these functions and have a list of all the module configs keyed by module name, and all parameters filled by the user was the core of this refactor, as all this information is required by other logic to come in this function.